src/groundlight/client.py

```diff
@@ -226,11 +285,14 @@ def submit_image_query(  # noqa: PLR0913 # pylint: disable=too-many-arguments
             image_query = self.get_image_query(iq_id)
 
         if wait:
-            threshold = self.get_detector(detector).confidence_threshold
-            image_query = self.wait_for_confident_result(image_query, confidence_threshold=threshold, timeout_sec=wait)
+            if confidence_threshold is None:
+                threshold = self.get_detector(detector).confidence_threshold
+            else:
+                threshold = confidence_threshold
+            image_query = self._wait_for_confident_result(image_query, confidence_threshold=threshold, timeout_sec=wait)
         return self._fixup_image_query(image_query)
 
-    def wait_for_confident_result(
+    def _wait_for_confident_result(
```

Comment:
Should we hide this method?  I think it’s quite reasonable that customers will want to use this directly.  We have a bunch ourselves, but a lot of that was because the `ask_` methods weren’t available.  But I think, e.g. in the ask_async example with two threads - one thread just submits with ask_async, and the other one uses wait_for_confident to analyze.

Reply:
It's more of a gut instinct that made me want to hide it originally. I suspect that any use case someone has for it we'll want to build syntactic sugar around it like we're doing for the ask methods. The ask_async method may be the exception, so I'll make this unhidden for now.